src/main/kotlin/com/demonwav/mcdev/platform/mixin/action/GenerateAccessorHandler.kt

```diff
@@ -0,0 +1,496 @@
+    }
+
+    override fun hasMembers(aClass: PsiClass): Boolean {
+        if (aClass.isMixin) {
+            return false
+        }
+
+        if (aClass.fields.any { canHaveAccessor(it) }) {
+            return true
+        }
+        if (aClass.methods.any { canHaveInvoker(it) }) {
+            return true
+        }
+
+        return false
+    }
+
+    override fun getAllOriginalMembers(aClass: PsiClass?): Array<ClassMember> {
+        if (aClass == null) {
+            return ClassMember.EMPTY_ARRAY
+        }
+
+        val members = mutableListOf<ClassMember>()
+        members.addAll(
+            aClass.fields
+                .filter { canHaveAccessor(it) }
+                .map { PsiFieldMember(it) }
+        )
+        members.addAll(
+            aClass.methods
+                .filter { canHaveInvoker(it) }
+                .map { PsiMethodMember(it) }
+        )
+        return members.toTypedArray()
+    }
+
+    override fun chooseOriginalMembers(aClass: PsiClass?, project: Project?, editor: Editor?): Array<ClassMember>? {
+        project ?: return null
+
+        val offset = editor?.caretModel?.offset ?: return null
+        val element = aClass?.containingFile?.findElementAt(offset) ?: return null
+        val targetClass = element.findContainingClass() ?: return null
+        val candidates = getAllOriginalMembers(targetClass)
+
+        val headerPanel = HeaderPanel()
+        val chooser = MemberChooser(candidates, false, true, project, headerPanel, arrayOf())
+        chooser.selectElements(
+            candidates.filter { candidate ->
+                if (candidate !is PsiElementClassMember<*>) {
+                    return@filter false
+                }
+                val range = candidate.element.textRange
+                return@filter range != null && range.contains(offset)
+            }.toTypedArray()
+        )
+
+        if (!chooser.showAndGet()) {
+            return null
+        }
+        val selectedMembers = chooser.selectedElements
+        if (selectedMembers.isNullOrEmpty()) {
+            return null
+        }
+
+        generateGetters = headerPanel.gettersCheckbox.isSelected
+        generateSetters = headerPanel.settersCheckbox.isSelected
+
+        return selectedMembers.toTypedArray()
+    }
+
+    private fun canHaveAccessor(field: PsiField): Boolean {
+        val isPublic = field.modifierList?.hasExplicitModifier(PsiModifier.PUBLIC) == true
+        val isFinal = field.modifierList?.hasExplicitModifier(PsiModifier.FINAL) == true
+        val isEnumConstant = field is PsiEnumConstant
+        return (!isPublic || isFinal) && !isEnumConstant
+    }
+
+    private fun canHaveInvoker(method: PsiMethod): Boolean {
+        return !method.modifierList.hasExplicitModifier(PsiModifier.PUBLIC)
+    }
+
+    private fun getOrCreateAccessorMixin(project: Project, targetClass: PsiClass): PsiClass? {
+        val mixins = MixinModule.getAllMixins(project, GlobalSearchScope.projectScope(project))
```

Comment:
Later it might be nice to store Mixins (and accessor Mixins) in a cache to speed this up. It's not slow when testing in SpongeForge, but for me it seemed to hang for half a second. Nothing major, just something to consider for later.